hoag/lbfgs.py

```diff
@@ -75,8 +74,24 @@ def lbfgs(
     grad_x = f_grad(x)
 
     y_list, s_list, mu_list = [], [], []
+    if warm_restart_lists is not None:
+        s_list, y_list, mu_list = warm_restart_lists
+    t = [1 / k for k in range(1, max_iter + 1)]
     for k in range(1, max_iter + 1):
-
+        if inverse_direction_fun is not None and k%inverse_secant_freq == 0 and k > 1:
+            # update the memory with the extra secant condition for inverse
+            inverse_direction = inverse_direction_fun(x)
+            e = -t[k-1] * two_loops(inverse_direction, m, s_list, y_list, mu_list, B0)
```

Comment:
With the follow up scaling, this is not useful I guess
```suggestion
            e = - two_loops(inverse_direction, m, s_list, y_list, mu_list, B0)
```